server/api/db/session.py

```diff
@@ -37,3 +37,17 @@ def run_function_with_new_db_session(func, *args, **kwargs):
         return result
     finally:
         close_session(session)
+
+
+async def run_async_function_with_new_db_session(func, *args, **kwargs):
```

Comment:
I think this can be merged with the above using `maybe_coroutine`

Reply:
you can also use `run_function_with_new_db_session` that will return a future and then await on it